pkg/util/ec2/ec2.go

```diff
@@ -239,9 +239,10 @@ func doHTTPRequest(url string, method string, headers map[string]string, useToke
 }
 
 func getToken() (string, error) {
+
 	token.RLock()
-	// Will refresh token 15 seconds before expiration
-	if time.Now().Before(token.expirationDate.Add(-15 * time.Second)) {
+	// The token renewal window is open, refreshing the token
+	if time.Now().Before(token.expirationDate.Add(-1 * tokenRenewalWindow)) {
```

Comment:
Could we subtract that second when we set the expiration date rather than when we check if it's expired ?